[quant][graphmode] Skip inserting observers for Tensors inside fused op

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25403 [clang-format] torch/jit/passes/quantization.{h,cpp} and torch/jit/init.cpp
* #25370 [refactor] Remove unnecessary checks in InsertQuantDeQuantImpl
* **#25281 [quant][graphmode] Skip inserting observers for Tensors inside fused op**

Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17106037](https://our.internmc.facebook.com/intern/diff/D17106037)

jerryzh168 committed:
[quant][graphmode] Skip inserting observers for Tensors inside fused op

Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] Skip inserting observers for Tensors inside fused op"


Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] Skip inserting observers for Tensors inside fused op"


Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] Skip inserting observers for Tensors inside fused op"


Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] Skip inserting observers for Tensors inside fused op"


Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] Skip inserting observers for Tensors inside fused op"


Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

ZolotukhinM reviewed (approved):
Overall looks good! Please find a couple of remarks inline.

jerryzh168 committed:
Update on "[quant][graphmode] Skip inserting observers for Tensors inside fused op"


Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] Skip inserting observers for Tensors inside fused op"


Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17106037](https://our.internmc.facebook.com/intern/diff/D17106037)

jerryzh168 committed:
Update on "[quant][graphmode] Skip inserting observers for Tensors inside fused op"


Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17106037](https://our.internmc.facebook.com/intern/diff/D17106037)

jerryzh168 committed:
Update on "[quant][graphmode] Skip inserting observers for Tensors inside fused op"


Summary:
We want to skip inserting observers for the Tensors that's between the two
ops that will be fused, e.g. Conv -> ReLU, this PR just added this pattern,
but new patterns can be easily added in the future.

Test Plan:
python test test/test_jit.py -- 'TestJit.test_insert_observers_skip_values'
Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17106037](https://our.internmc.facebook.com/intern/diff/D17106037)
